### ISSUE TITLE
Mouse hover issue on menu bar when remotely controlling Mac (ipad->mac)

### ISSUE THREAD

**Issue by rustdesk** [rustdesk/rustdesk#8789]
### Discussed in https://github.com/rustdesk/rustdesk/discussions/8788

<div type='discussions-op-text'>

<sup>Originally posted by **ZhuJiaxin2** July 23, 2024</sup>
### Bug Description

When controlling Mac through my iPad, i use apple magic mouse to click on iPad. 
However, when my mouse hover over the menu bar area, the hover function could not work (e.g. my mouse clicks on the apple icon in the menu bar, and several selections were listed under, when my mouse hover downward to reach for the selections, the selections disapears), this issue does not appear in the middle of the screen, when my mouse right clicks and hovers for paste or copy selections.
Also, this has caused great inconvenience, for i could not remotely close my full screen window in Mac! (because of the hovering issue, when my mouse is up there, the close tab button does not appear!)
So grateful for your effort!

### How to Reproduce

I dunno

### Expected Behavior

in Mac, when my mouse click on a menu bar button, and then hover downward for option selection, the selections stay there instead of disappearing under the mouse.

### Operating system(s) on local side and remote side

iPad->Mac

### RustDesk Version(s) on local side and remote side

1.2.6->1.2.6

### Screenshots

Click on menu bar, selections came out:
<img width="1440" alt="截屏2024-07-23 上午11 30 16" src="https://github.com/user-attachments/assets/1a85f5a5-dbdc-46c2-b12e-cbab2dafdf31">
Expectation when mouse hover downward:
<img width="1440" alt="截屏2024-07-23 上午11 30 38" src="https://github.com/user-attachments/assets/cbba4ded-7d2b-4ee5-9a1d-31df0978936e">
What actually happens:
<img width="1440" alt="截屏2024-07-23 上午11 30 28" src="https://github.com/user-attachments/assets/dde962aa-66e2-4c69-ba51-6e2e90bd5c70">

The hidden close tab button:
<img width="1440" alt="截屏2024-07-23 上午11 40 00" src="https://github.com/user-attachments/assets/916c2cc2-f194-4141-b685-0ea86c0f0e74">
The close tab button only appears when mouse hovers there, but this doesnt work on this case, for the hovering function in that area is disabled
<img width="1440" alt="截屏2024-07-23 上午11 40 05" src="https://github.com/user-attachments/assets/4509ccc4-ff34-4926-bec4-c35a1ed2b894">

### Additional Context

_No response_</div>

**Comment by rustdesk** [comment]
Apologize for the wrong conversion. https://github.com/rustdesk/rustdesk/issues/8789

**Comment by rustdesk** [comment]
No issue if it is not menubar?

**Comment by ZhuJiaxin2** [comment]
> No issue if it is not menubar? @ZhuJiaxin2

yes, only happens on menu bar and close tab, hover on screen is normal:
<img width="1440" alt="截屏2024-07-23 上午11 57 52" src="https://github.com/user-attachments/assets/00427136-86cf-4f88-89db-e1bb24d2ffc2">

**Comment by EmranMR** [comment]
Work around is to press and hold for time being .

I have also noticed the mouse moves to top left, causing the top menu to come down and stage manager to show when the mouse is inactive for a period of time?

I have also noticed in some apps, the clicking and double-clicking just not work all together.
https://github.com/user-attachments/assets/03fbd872-461f-4a00-bc99-d4421a1c3e2a

**Comment by lakie0605** [comment]
When controlling with a mobile device, every time the cursor is moved, a "left click" is performed at the original cursor position, which is very annoying.

**Comment by mxioi** [comment]
I've noticed this issue as well and raised a new bug here: #12388

**Comment by chriskhoffman** [comment]
I have been running into this bug as well, and would love to see a fix for it.

I have been using the click and hold workaround for the last year, but I recently updated to a new iPad OS version and now that workaround is no longer viable. In the new OS version, clicking and holding at the top of the screen now triggers a new Stage Manager behavior where it tries to drag/resize the app window.

Additional info: I am using iPad Pro M4 13" on iPad OS 26.0 (which is a preview/beta version), and using Rustdesk iPad app version 1.4.0 to connect to a MacBook Pro using app service version 1.4.1.

When this new iPad OS version goes public, this will be a much bigger problem.

**Comment by mxioi** [comment]
Only tested once so may not work for others but just discoverd today under the display options in the app theres a option to add a virtual display. Doing this has seemed to fix a few issue ive had viruses not having a virtual display. Intrested to see if this is a one off and if this fix works for others?

**Comment by junwha** [comment]
@mxoio Could you clarify where can we find the virtual display option? I cannot find it on ios 18.7.1

**Comment by mxioi** [comment]
> @mxoio Could you clarify where can we find the virtual display option? I cannot find it on ios 18.7.1

![image](https://github.com/user-attachments/assets/87cf9c12-e29e-4ea0-a612-61889e8f5a20)

![image](https://github.com/user-attachments/assets/4e9795a0-2cb3-46ff-ae2d-44c33e6f8740)

![image](https://github.com/user-attachments/assets/df9cba70-879f-4764-9216-a92da12f71c9)

![image](https://github.com/user-attachments/assets/8f660d6e-a9e1-453d-b4b9-6f37935e17c6)


Of course your only need one virtual display driver but that works a lot better. A lot less issues, not perfect, every now again I have to login off and on but a lot better than before. Hope this helps!


@rustdesk maybe helpful if you do get any future tickets regarding this issue.

**Comment by rustdesk** [comment]
Is this specific to iOS? Android does not have this issue? Frankly, I do not understand this issue quite clearly. If any one can record a video?

**Comment by rustdesk** [comment]
Here is a fix proposed by Copilot. No idea if who can help build and test. Our team do not have time yet.

**Comment by mxioi** [comment]
> Is this specific to iOS? Android does not have this issue? Frankly, I do not understand this issue quite clearly. If any one can record a video?

I can’t confirm whether this affects Android, but the issue does appear to be iOS-specific. It seems related to how touch gestures and external peripherals are interpreted once they pass through to Windows. My suspicion is that part of the problem comes from Windows’ own input drivers being unreliable in this scenario, which is supported by the fact that the issue is partially resolved when a virtual input driver is applied and that virtual display is used instead.


I can't upload the video directly to GitHub as it's to large of a file or just can't compress for some reason 

For more details I did note the bug behaviour bellow which has a google drive link where I have uploaded the video of the issue

#12388

**Comment by mxioi** [comment]
> Here is a fix proposed by Copilot. No idea if who can help build and test. Our team do not have time yet.

This might be very difficult to fix because of how input from different devices is translated, as well as the general instability of Windows’ touch-related drivers. The issue seems to come from the way touch events are converted into mouse-style input once they reach Windows. Someone with deeper knowledge of the input pipeline may be able to address it, but for now the virtual display driver does mitigate the problem. If anyone wants to investigate further, comparing the behaviour with and without the virtual display driver enabled might help reveal where the translation breaks down. Unfortunately, I’m not experienced enough with driver or input-handling code to dig into this myself.

**Comment by junwha** [comment]
Thanks @mxoio! but looks I don't have that menu probably due to the lower ios version.

@rustdesk I moved to android due to this issue, and I can confirm that android does not have this problem. but today I tested my iphone again, and the same issue still occurs (**the cursor moves to the left top at every click on any place**) I had explained my observation here: [Discussion#10270](https://github.com/rustdesk/rustdesk/discussions/10270).

I just wonder if there is any ios device or version that rustdesk with mouse works correctly. Thanks.
